File: 1411-convert-binary-number-in-a-linked-list-to-integer/solution.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    int getDecimalValue(ListNode* head) {
        
        ListNode* n = head ;
        int res = 0 ;
        while(n != NULL){
            
            // res = res*2 + n->val;
            // n=n->next ;
            
            res <<= 1 ;
            res += n->val ;
            n = n ->next ; 
            
        }
        return res ;
    }
};

File: 1411-convert-binary-number-in-a-linked-list-to-integer/solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public int getDecimalValue(ListNode head) {

        ListNode temp = head ;
        List<Integer> list = new ArrayList<>();
        while(temp != null){
            list.add(temp.val);
            temp = temp.next;
        }
        Collections.reverse(list);
        int k = 0 , res = 0 ;
        for(Integer i : list){
            res += (int)i*Math.pow(2,k++);
        }
        return res ;
    }
}
